primitives/file-key-verifier/src/lib.rs

```diff
@@ -71,7 +71,7 @@ where
     /// and checks if the resulting leaf is in the proof.
     fn verify_proof(
         expected_file_key: &Self::Commitment,
-        challenges: &[Self::Challenge],
+        challenges: &[(Self::Challenge, Option<ChallengeKeyInclusion>)],
```

Comment:
We probably don't need to add this here since there will probably be no case for checking a chunk being included or unincluded in a proof. We simply check if the challenge has been responded to appropriately with a _valid_ (based on our proof verifying algorithm) proof.

Reply:
I would try to not change this interface if possible. If this can be handled by the upper layer using it, we should keep this simpler.

Reply:
The only thing I changed was the return type from Vec to BTreeSet